aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -210,9 +216,8 @@
   public static final String CLIENT_ASSUME_ROLE_REGION = "client.assume-role.region";
 
   /**
-   * Enables eTag checks for S3 PUT and MULTIPART upload requests.
+   * @deprecated will be removed at 0.15.0, please use {@link #S3_CHECKSUM_ENABLED_DEFAULT} instead
    */
-  public static final String S3_CHECKSUM_ENABLED = "s3.checksum-enabled";
   public static final boolean CLIENT_ENABLE_ETAG_CHECK_DEFAULT = false;
```

Comment:
[minor] we might need to add `@Deprecated` here as well.

Reply:
good point, adding